packages/smooth_app/lib/pages/user_management/sign_up_page.dart

```diff
@@ -0,0 +1,311 @@
+              subtitle: !_disagreed
+                  ? null
+                  : Text(
+                      'You have to agree my friend!',
+                      style: TextStyle(color: Theme.of(context).errorColor),
+                    ),
+            ),
+            const SizedBox(height: space),
+            ListTile(
+              leading: Checkbox(
+                value: _foodProducer,
+                onChanged: (final bool? value) {
+                  if (value != null) {
+                    setState(() => _foodProducer = value);
+                  }
+                },
+              ),
+              title: Text('I am a food producer'),
+            ),
+            if (_foodProducer) ...<Widget>[
+              const SizedBox(height: space),
+              SmoothTextFormField(
+                type: TextFieldTypes.PLAIN_TEXT,
+                controller: _brandController,
+                hintText: 'Producer/brand',
+                prefixIcon: const Icon(Icons.person),
+                autofillHints: const <String>[AutofillHints.name],
+                validator: (String? value) {
+                  if (value == null || value.isEmpty) {
+                    return 'Please enter a producer or a brand name';
+                  }
+                  return null;
+                },
+              )
+            ],
+            const SizedBox(height: space),
+            ListTile(
+              leading: Checkbox(
+                value: _subscribe,
+                onChanged: (final bool? value) {
+                  if (value != null) {
+                    setState(() => _subscribe = value);
+                  }
+                },
+              ),
+              title: Text(
+                  'I\'d like to subscribe to the Open Food Facts newsletter (Note: I can unsubscribe from it at all time)'),
+            ),
+            const SizedBox(height: space),
+            ElevatedButton(
+              onPressed: () async => _signUp(),
+              child: Text(
+                'Sign Up',
+                style: theme.textTheme.bodyText2?.copyWith(
+                  fontSize: 18.0,
+                  color: theme.colorScheme.surface,
+                ),
+              ),
+              style: ButtonStyle(
+                minimumSize: MaterialStateProperty.all<Size>(
+                  Size(size.width * 0.5, theme.buttonTheme.height + 10),
+                ),
+                shape: MaterialStateProperty.all<RoundedRectangleBorder>(
+                  RoundedRectangleBorder(
+                    borderRadius: BorderRadius.circular(300.0),
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> _signUp() async {
+    _disagreed = !_agree;
+    setState(() {});
+    if (!_formKey.currentState!.validate() || _disagreed) {
+      return;
+    }
+    _popEd = false;
+    await _openSigningUpDialog();
+    // TODO(monsieurtanuki): then what?
```

Comment:
We probably need a next param, so that we can integrate signup in various flows (like signup to contribute from the product page, where we'd apply the contribution with the newly created account/credentials)

Reply:
@teolemon I don't think we need a next param. Whoever calls the sign up page can `await` it and do whatever afterwards, according to the `pop`'ed result.

Reply:
And I confirm: `SignUpPage` now pop's `true` when successful. Just check the return code and do what you want with it.